src/smc-webapp/course/settings_panel.cjsx

```diff
@@ -0,0 +1,724 @@
+        content = "# Course '#{@props.settings.get('title')}'\n"
+        content += "# exported #{timestamp}\n"
+        content += "Name,Email,"
+        content += ("\"#{assignment.get('path')}\"" for assignment in assignments).join(',') + '\n'
+        for student in store.get_sorted_students()
+            grades = ("\"#{store.get_grade(assignment, student) ? ''}\"" for assignment in assignments).join(',')
+            name   = "\"#{store.get_student_name(student)}\""
+            email  = "\"#{store.get_student_email(student) ? ''}\""
+            line   = [name, email, grades].join(',')
+            content += line + '\n'
+        @write_file(@path('csv'), content)
+
+    save_grades_to_py : ->
+        ###
+        example:
+        course = 'title'
+        exported = 'iso date'
+        assignments = ['Assignment 1', 'Assignment 2']
+        students=[
+            {'name':'Foo Bar', 'email': 'foo@bar.com', 'grades':[85,37]},
+            {'name':'Bar None', 'email': 'bar@school.edu', 'grades':[15,50]},
+        ]
+        ###
+        timestamp = (salvus_client.server_time()).toISOString()
+        store = @props.redux.getStore(@props.name)
+        assignments = store.get_sorted_assignments()
+        students = store.get_sorted_students()
+        content = "course = '#{@props.settings.get('title')}'\n"
+        content += "exported = '#{timestamp}'\n"
+        content += "assignments = ["
+        content += ("'#{assignment.get('path')}'" for assignment in assignments).join(',') + ']\n'
+
+        content += 'students = [\n'
+        for student in store.get_sorted_students()
+            grades = (("'#{store.get_grade(assignment, student) ? ''}'") for assignment in assignments).join(',')
+            name   = store.get_student_name(student)
+            email  = store.get_student_email(student)
+            email  = if email? then "'#{email}'" else 'None'
+            line   = "    {'name':'#{name}', 'email':#{email}, 'grades':[#{grades}]},"
+            content += line + '\n'
+        content += ']\n'
+        @write_file(@path('py'), content)
+
+    render_save_grades : ->
+        <Panel header={@render_grades_header()}>
+            <div style={marginBottom:'10px'}>Save grades to... </div>
+            <ButtonToolbar>
+                <Button onClick={@save_grades_to_csv}><Icon name='file-text-o'/> CSV file...</Button>
+                <Button onClick={@save_grades_to_py}><Icon name='file-code-o'/> Python file...</Button>
+            </ButtonToolbar>
+            <hr/>
+            <span style={color:"#666"}>
+                Export all the grades you have recorded
+                for students in your course to a csv or Python file.
+            </span>
+        </Panel>
+
+    ###
+    # Help box
+    ###
+    render_help : ->
+        <Panel header={<h4><Icon name='question-circle' />  Help</h4>}>
+            <span style={color:"#666"}>
+                <ul>
+                    <li>
```

Comment:
the first thing I looked at is, if you have merged. I've added a link to this intro by mike croucher which isn't here. I fear I should have told you, that I've added a link :-\

https://github.com/sagemathinc/smc/commit/c7b7404c1011c768eb6ba09756c73d10e5f2b690